app/components/SaveControl/SaveControl.js

```diff
@@ -76,8 +76,10 @@ class SaveControl extends Component {
     }
   }
 
-  handleToggle = (e, state) => {
-    if (state) {
+  handleToggle = () => {
+    if (this.props.article.get('id')) {
```

Comment:
I don't think we should fetch article there. While the fetched data is more coherent with the backend, that may be not what user expects.

Imagine next situation: you draw the popup and the toggle is off. The user clicks toggle (expecting the article to be saved), but the backend reports the article is already saved now, so we delete it here.